zebra-state/src/service.rs

```diff
@@ -209,6 +202,21 @@ impl StateService {
             for (child, rsp_tx) in queued_children {
                 let child_hash = child.hash;
                 tracing::trace!(?child_hash, "validating queued child");
+
+                // required by validate_and_commit
+                let mandatory_checkpoint = self.network.mandatory_checkpoint_height();
+                if child.height <= mandatory_checkpoint {
+                    panic!(
+                        "invalid non-finalized block height: the canopy checkpoint is mandatory, pre-canopy blocks, and the canopy activation block, must be committed to the state as finalized blocks"
+                    );
```

Comment:
Nit: Splitting the line just to make reading it easier.

```suggestion
                    panic!(
                        "invalid non-finalized block height: the canopy checkpoint is mandatory, \
                        pre-canopy blocks, and the canopy activation block, must be committed to \
                        the state as finalized blocks"
                    );
```

Reply:
Fixed in ca02edac.

I also improved the other assertion that was moved, and the surrounding comments.